CMakeLists.txt

```diff
@@ -4,14 +4,9 @@ project( EOSIO )
 include(CTest) # suppresses DartConfiguration.tcl error
 enable_testing()
 
-if ("${CMAKE_INSTALL_PREFIX}" STREQUAL "/usr/local")
-   message(WARNING "CMAKE_INSTALL_PREFIX is set to /usr/local. This is not recommended.")
-endif()
-
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
 
 include( GNUInstallDirs )
-include( InstallDirectoryPermissions )
```

Comment:
Wouldn't that have affected files too? This function was _only_ was dealing with directories and I don't see anything else that would have solved this w.r.t. files.

Reply:
It did, and I think the eventual solution was to manually control umask in CI/CD, where the packages were built.  If that's our preference, rather than overriding CMake's arguably broken behavior, then we should document it.

Reply:
My impression on a very limited experiment was that the issue is more nuanced than what you're describing and the original bug entry is still correct in its limited scope:

> wrong permission on base directories

That is, when `install(FILES` in to a/b/c/ any of those directories (a, b, or c) will be created with current umask unexpectedly applied. Many other install directives, like even the actual files in that previous case, or the target directory in `install(DIRECTORY` even, don't suffer from the same problem. As an example, chainbase does not have this macro but its `install(DIRECTORY` creates the chainbase directory with the expected permissions ignoring the current umask.

afaict this is neglected cruft. The license files being a perfect example of something that never had the workaround applied for years and were subject to restrictive umasks. and if you do `cmake -DCMAKE_INSTALL_PREFIX=/opt/eosio` and then `make install` you can end up with stuff like
```
$ ls -l /opt/eosio
total 0
drwx------ 1 root root 116 May 23 16:06 bin
drwx------ 1 root root  10 May 23 16:06 etc
drwx------ 1 root root 114 May 23 16:06 include
drwx------ 1 root root 346 May 23 16:06 lib
drwx------ 1 root root  28 May 23 16:06 share
drwx------ 1 root root  12 May 23 16:06 var
```
So trying to fight against cmake's behavior just looks like a losing battle and not worth the trade offs.

Reply:
We will continue to fight against CMake's misbehavior.  The permissions shown in `/opt/eosio` are completely unacceptable.  But let's revisit the fight in another PR.